cf-keycrypt/cf-keycrypt.c

```diff
@@ -0,0 +1,388 @@
+/*
+   cf-keycrypt.c
+
+   Copyright (C) 2017 cfengineers.net
+
+   Written and maintained by Jon Henrik Bjornstad <jonhenrik@cfengineers.net>
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+#include <config.h>
+#include <cf3.defs.h>
+#include <cf3.extern.h>
+#include <openssl/err.h>
+
+#include <lastseen.h>
+#include <conversion.h>
+#include <files_hashes.h>
+#include <locks.h>
+#include <item_lib.h>
+#include <known_dirs.h>
+#include <dbm_api.h>
+/*
+#ifdef LMDB
+#include <lmdb.h>
+#endif
+*/
+
+#define STDIN     0
+#define STDOUT    1
+
+#define BUFSIZE 1024
+
+void usage()
+{
+    printf("Workdir is defined as %s\n", WORKDIR);
+    printf(
+    "\n"
+    "Usage: cf-keycrypt [-e public-key|-d private-key|-H hostname-or-ip] -o outfile -i infile [-h]\n"
```

Comment:
Apparently the command needs more options than my previous examples. It needs a key file OR a hostname/IP. 
* `-e` or `--encrypt`
* `-d` or `--decrypt`
* `-k` or `--keyfile`
* `-H` or `--host`
* `-o` or `--output`
*  `filename` should be a mandatory argument

It's important to agree to a sensible command line usage *before* merging so that we don't break things later.